src/c++/perf_analyzer/report_writer.cc

```diff
@@ -235,6 +246,14 @@ ReportWriter::GenerateReport()
           }
         }
       }
+      if (should_output_llm_metrics_) {
+        if (collector_->IsEmpty()) {
```

Comment:
nit: I think it'd be ok to put this logic closer to where the `collector_` is used, e.g. in `CalculateLlmMetrics()`